src/Microsoft.Health.Fhir.Core/Features/Operations/DataConvert/FhirServerBuilderDataConvertRegistrationExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Operations.DataConvert;
+using Microsoft.Health.Fhir.Core.Registration;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class FhirServerBuilderDataConvertRegistrationExtensions
+    {
+        public static IFhirServerBuilder AddDataConvert(this IFhirServerBuilder fhirServerBuilder)
+        {
+            EnsureArg.IsNotNull(fhirServerBuilder, nameof(fhirServerBuilder));
+
+            fhirServerBuilder.AddDataConvertTemplateProvider()
+                .AddDataConvertEngine();
+
+            return fhirServerBuilder;
+        }
+
+        private static IFhirServerBuilder AddDataConvertEngine(this IFhirServerBuilder fhirServerBuilder)
+        {
+            EnsureArg.IsNotNull(fhirServerBuilder, nameof(fhirServerBuilder));
```

Comment:
We don't have to do arg checking for private members where the argument has already been checked in a public member

Reply:
Agree. I have removed these checks in private function